x-pack/legacy/plugins/uptime/public/components/functional/condensed_check/to_condensed_check.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { get } from 'lodash';
+import { Check } from '../../../../common/graphql/types';
+import { CondensedCheck } from '../monitor_list';
+
+const inferCondensedFields = (
+  check: CondensedCheck,
+  currentStatus: string,
+  currentTimestamp: string
+) => {
+  const { status: condensedStatus, timestamp } = check;
+  if (condensedStatus !== currentStatus && condensedStatus !== 'mixed') {
+    check.status = 'mixed';
+  }
+  if (timestamp < currentTimestamp) {
+    check.timestamp = currentTimestamp;
+  }
+};
+
+export const toCondensedCheck = (checks: Check[]) => {
+  const condensedChecks: Map<string | null, CondensedCheck> = new Map();
+  checks.forEach((check: Check) => {
+    const location = get<string | null>(check, 'observer.geo.name', null);
+    const {
+      monitor: { ip, status },
+      timestamp,
+    } = check;
+    let condensedCheck: CondensedCheck | undefined;
+    if ((condensedCheck = condensedChecks.get(location))) {
+      condensedCheck.childStatuses.push({ ip, status });
```

Comment:
We should include `timestamp` here as well.

Reply:
Resolved in https://github.com/elastic/kibana/pull/38895/commits/7bc068c01b84090fd328698688ab3c3fcdb6818e and  https://github.com/elastic/kibana/pull/38895/commits/0878df98f32539daa7892d223dfa35ebf9d0abac.